### DISCUSSION TITLE
Prisma upserts non-deterministically throw `Unique constraint failed on the fields` errors

### DISCUSSION THREAD

**Discussion by morleyzhi** [prisma/prisma#16583] [Q&A]
In several places, our code either upserts or finds ? updates : creates a settings value, which we find via setting key + owner id instead of id. But a very small percentage of the time, Prisma returns a "Unique constraint failed" error on the create / update / upsert function. It seems to imply that even in the upsert call, the database row doesn't exist when checked, but exists when updated. Which seems impossible!

It seems even more perplexing since it seems to happen randomly and rarely, so I can't really create a test case. Has anyone seen this before?

I'm running a Next.js app on Vercel, connected to a Postgresql database with connection pooling. I'm wondering if this is a connection pool issue, because it makes no sense?

**Comment by pantharshit00** [comment]
Hello @morleyzhi,

If you are using Postgres, can you try the latest release and see if this is happening with that as well. We have changed underlying code for upserts in https://github.com/prisma/prisma/releases/tag/4.6.0 release.

If this is still occurring with the latest version, do let me know and please provide a full error stacktrace if possible.

**Reply by vladi-stevanovic** [reply]
👋 Hi @morleyzhi, did you perhaps have a chance to check Harshit's recommendation? Let us know if you still have any questions! 

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by morleyzhi** [comment]
Hi, we haven’t been able to upgrade yet because versions after 4.4.0 don’t work when imported as commonjs modules, which breaks our ts-node scripts. I’m hoping to be able to look into it next week or later.

**Reply by nurul3101** [reply]
Do let us know in case you still get an error after upgrading to the latest Prisma version.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
